config/plugins/poe.py

```diff
@@ -0,0 +1,213 @@
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    try:
+        cfg = sonic_cfggen.FormatConverter.to_serialized(copy.deepcopy(cfg))
+        config_mgmt.ConfigMgmt().loadData(cfg)
+    except Exception as err:
+        raise Exception('Failed to validate configuration: {}'.format(err))
+
+
+def _update_entry_validated(db, table, key, data, create_if_not_exists=False):
+    """ Update entry in table and validate configuration.
+    If attribute value in data is None, the attribute is deleted.
+
+    Args:
+        db (swsscommon.ConfigDBConnector): Config DB connector obect.
+        table (str): Table name to add new entry to.
+        key (Union[str, Tuple]): Key name in the table.
+        data (Dict): Entry data.
+        create_if_not_exists (bool):
+            In case entry does not exists already a new entry
+            is not created if this flag is set to False and
+            creates a new entry if flag is set to True.
+    Raises:
+        Exception: when cfg does not satisfy YANG schema.
+    """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+
+    if not data:
+        raise Exception(f"No field/values to update {key}")
+
+    if create_if_not_exists:
+        cfg[table].setdefault(key, {})
+
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not have PoE configuration")
+
+    entry_changed = False
+    for attr, value in data.items():
+        if value == cfg[table][key].get(attr):
+            continue
+        entry_changed = True
+        if value is None:
+            cfg[table][key].pop(attr, None)
+        else:
+            cfg[table][key][attr] = value
+
+    if not entry_changed:
+        return
+
+    _validate_config_or_raise(cfg)
+    db.set_entry(table, key, cfg[table][key])
+
+
+# 'poe' subcommand ("config poe ...")
+@click.group(
+    name="poe",
+    cls=clicommon.AliasedGroup,
+)
+def poe():
+    """  Configure PoE (Power over Ethernet) feature """
+    pass
+
+
+# 'interface' subcommand ("config poe interface ...")
+@poe.group(
+    name="interface",
+    cls=clicommon.AliasedGroup,
+)
+def poe_interface():
+    """  Configure PoE interface """
+    pass
+
+
+# 'status' subcommand ("config poe interface status ...")
+@poe_interface.command(
+    name="status"
+)
+@click.argument(
+    "ifname",
+    nargs=1,
+    required=True,
+)
+@click.argument(
+    "enabled",
+    nargs=1,
+    required=True,
+    type=click.Choice(["enable", "disable"])
+)
+@clicommon.pass_db
+def poe_intf_status(db, ifname, enabled):
+    """  Enable or disable PoE on interface """
+
+    data = {}
+    if enabled is not None:
+        data["enabled"] = enabled
+
+    try:
+        _update_entry_validated(db.cfgdb, POE_PORT, ifname, data)
+    except Exception as err:
+        _exit_with_error(f"Error: {err}", fg="red")
+
+
+# 'power-limit' subcommand ("config poe interface power-limit ...")
+@poe_interface.command(
+    name="power-limit"
```

Comment:
Please clarify if this CLIs will be available only in supported platforms

Reply:
The CLI will be available on all platforms. If the platform does not support PoE the user will get an error message when trying to configure anything on an interface.